pretyping/glob_ops.ml

```diff
@@ -577,6 +577,10 @@ let glob_constr_of_closed_cases_pattern env p = match DAst.get p with
 
 let glob_constr_of_cases_pattern env p = glob_constr_of_cases_pattern_aux env false p
 
+let glob_term_printer = ref (fun _ _ -> failwith "Glob printer not installed")
+let print_glob_constr_env env t = !glob_term_printer env t
+let set_print_glob_constr f = glob_term_printer := f
+
```

Comment:
I'd suggest we don't add an imperative hook but we instead define a proper error with is raised and handled in the UI layer.

Reply:
I could do that. My worries are that, only say for the `interp` directory, there are 93 `user_err`. I know by experience that I'm sometimes refraining to use errors referring to terms precisely because of the absence of printers directly available (and the error which I just changed in `reserve.ml` is of this kind). So, virtually, there are not 93 but still a couple of cases where we would like errors to be able to print terms. I'm afraid that adding them one by one would virtually give a big list. We could do like we do for `type_errors.ml` and `pretype_errors.ml`: add a new, say, `interp_errors.ml` to have both a control on errors and keep error matching in constant time . What do you think?

Or maybe add a "constr" node in `Pp.t`?

Reply:
I'm in favor of a proper error registration mechanism, as discussed in https://github.com/coq/coq/issues/7560

IMHO we have too many errors and adding an error is too easy for developers, but very hard for people trying to document them, UI support, etc...

Reply:
> I'm in favor of a proper error registration mechanism, as discussed in #7560

The warning infrastructure is very nice and applying it to errors too would be great.

Reply:
The discussion on errors would require more time, so I removed the corresponding commits. I can make another PR on this part, so that a discussion on what we could technically do, if ok for you.